sample/app-ios/app-ios/CustomNavigationView.swift

```diff
@@ -0,0 +1,167 @@
+import SwiftUI
+import Shared
+
+struct CustomNavigationView: View {
+
+    typealias Children = CustomNavigationComponentChildren<AnyObject, KittenComponent>
+
+    private let component: CustomNavigationComponent
+
+    @ObservedObject
+    private var observableChildren: ObservableValue<Children>
+
+    private var mode: CustomNavigationComponentMode {
+        observableChildren.value.mode
+    }
+
+    init(_ component: CustomNavigationComponent) {
+        self.component = component
+        observableChildren = ObservableValue(component.children)
+    }
+
+    var body: some View {
+        ZStack {
+            ChildItems(component, observableChildren.value)
+            VStack {
+                Spacer()
+                Buttons(component, mode).padding()
+            }
+        }
+    }
+}
+
+extension CustomNavigationView {
```

Comment:
Why do we need this extension? Can the underlying structs just be top-level and private?

Reply:
Agree, made them private and top-level